app/tests/heartbeat_controller_desktop_test.cc

```diff
@@ -56,6 +60,8 @@ class HeartbeatControllerDesktopTest : public ::testing::Test {
     storage_.Write(empty_heartbeats_struct);
     // Default to registering a user agent with version 1
     app_common::RegisterLibrariesFromUserAgent(kDefaultUserAgent);
+    // Override the min time between fetches to 0s
+    g_min_time_between_fetches_sec = 0.0;
```

Comment:
Overriding this for the entire test is a bit dangerous. We should only override it for the test which requires the override.

Here is my recommendation:
1. Add a variable to `HeartbeatControllerDesktopTest` class to store the original value of `g_min_time_between_fetches_sec`. 
2. Use `SetUp()` to always change it back to the original value. Also reset `last_read_all_heartbeats_time_` so that we can make sure the first fetch in any test works properly.
3. Only override `g_min_time_between_fetches_sec` inside the tests. Set to '0' only for tests which will call fetch more than once, ex.`GetTodaysHeartbeatThenGetAllHeartbeats`. Or set to to the test value like `GetHeartbeatsPayloadTimeBetweenFetches` to test the cooldown.

Reply:
I now reset the value to the original value in the test's TearDown to make the change less permanent.
Resetting in setup doesn't make sense to me since at that point I dont see how the test would know what to reset to.
last_read_all_heartbeats_time_ should not need to be reset since every test creates a new instance of the controller so it should take a default value of 0.

Reply:
As suggested in the chat, I have now updated it so that the override and undo is part of the test case, and it is only done for the test cases that perform multiple fetches.